spyder/plugins/plots/widgets/figurebrowser.py

```diff
@@ -634,6 +637,15 @@ def set_figureviewer(self, figure_viewer):
         """Set the bamespace for the FigureViewer."""
         self.figure_viewer = figure_viewer
 
+    def eventFilter(self, widget, event):
+        """
+        An event filter to trigger an update of the thumbnails size so that
+        their width fit that of the scrollarea.
+        """
+        if event.type() == QEvent.Resize:
+            self._update_thumbnail_size()
+        return super().eventFilter(widget, event)
```

Comment:
This syntax is incorrect in Python 2. That's what's making our tests fail in that version.